eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/BackPropagatedRSWriter.java

```diff
@@ -0,0 +1,259 @@
+                Update updateRSInOriginalAssembly = Update.update(RS_ATTRIBUTE, rsToAssign);
+                bulkSVEUpdates.updateOne(queryToLookUpSSHash, updateRSInOriginalAssembly);
+                clusteringCounts.addSubmittedVariantsClustered(1);
+
+                bulkSVOEInserts.insert(getUpdateOperation(submittedVariantEntity, rsToAssign));
+                clusteringCounts.addSubmittedVariantsUpdateOperationWritten(1);
+            }
+        }
+
+        handleClusteredVariantInserts(evaCVEInserts, dbsnpCVEInserts, numEVARSInserts, numDbsnpRSInserts);
+
+        if (numEvaRSAssignments > 0) {
+            evaSVEUpdates.execute();
+            evaSVOEInserts.execute();
+        }
+
+        if (numDbsnpRSAssignments > 0) {
+            dbsnpSVEUpdates.execute();
+            dbsnpSVOEInserts.execute();
+        }
+    }
+
+    private void handleClusteredVariantInserts(BulkOperations evaCVEInserts, BulkOperations dbsnpCVEInserts,
+                                               int numEVARSInserts, int numDbsnpRSInserts) {
+        try {
+            if (numEVARSInserts > 0) {
+                evaCVEInserts.execute();
+            }
+
+            if (numDbsnpRSInserts > 0) {
+                dbsnpCVEInserts.execute();
+            }
+        }
+        catch (DuplicateKeyException duplicateKeyException) {
+            MongoBulkWriteException writeException = ((MongoBulkWriteException) duplicateKeyException.getCause());
+            // Consider a scenario where we are back-propagating rs ID:
+            // RS1 corresponding to SS1 from ASM2 (remapped assembly) to ASM1 (original assembly)
+            // If the hash for back-propagated RS1 already exists in ASM1,
+            // it means that, for some reason, SS1 was not clustered with RS1 when ASM1 underwent clustering.
+            // It is better to flag such issues for later analysis.
+            extractUniqueHashesForDuplicateKeyError(writeException)
+                    .forEach(hash -> logger.error("Attempted to insert RS record with already existing hash " + hash +
+                                                         "during RS ID back-propagation! This could be symptomatic " +
+                                                         "of clustering issues with the original assembly "
+                                                         + this.originalAssembly + "."));
+        }
+    }
+
+    private ClusteredVariantEntity getSuitableRSFromRemappedSS(
+            Map<Long, List<SubmittedVariantEntity>> ssInRemappedAssemblyGroupedByID,
+            SubmittedVariantEntity submittedVariantEntity) {
+        // Currently if there is more than one entry for the SS accession in the remapped assembly,
+        // we choose the minimum RS ID as a hack since we don't have a way to map between the
+        // remapped SS and the original SS in such cases
+        // TODO: Re-visit during EVA-2612
```

Comment:
I'm not actually sure where these cases are coming from:
You've reported that different submitted variants can have the same SSid when the remapping has created new alleles (SS split that we do not handle), but these events will have the same RSid (if they have any) as well so you will not separate them here.

Reply:
@tcezard Remapping is only one source of multiple SS IDs. Since we don't impose strict SS ID uniqueness in the dbSNP SS collection, it is possible to have the same SS ID with different RS due to other reasons (ex: different types of variants like SNV and INDEL in the same position) in the _source_ assembly. I will revise the comment to reflect this.